render/wgpu/src/target.rs

```diff
@@ -186,9 +188,26 @@ impl TextureTarget {
                         .extend_from_slice(&chunk[..self.buffer_dimensions.unpadded_bytes_per_row]);
                 }
 
-                let image = image::RgbaImage::from_raw(self.size.width, self.size.height, buffer);
+                let mut image =
+                    image::RgbaImage::from_raw(self.size.width, self.size.height, buffer);
                 drop(map);
                 self.buffer.unmap();
+
+                // Our texture uses premutliplied alpha - convert to straight alpha
+                // so that this image can be saved directly as a PNG. Note:
+                // the `image` library doesn't appear to track whether or
+                // not an image uses premultiplied or straight alpha.
+                if let Some(image) = &mut image {
+                    for pixel in image.pixels_mut() {
+                        let scale = pixel[3] as f64 / 255.0;
+                        *pixel = image::Rgba([
+                            (pixel[0] as f64 / scale) as u8,
+                            (pixel[1] as f64 / scale) as u8,
+                            (pixel[2] as f64 / scale) as u8,
+                            pixel[3],
+                        ]);
```

Comment:
Should probably handle the alpha = 0 case to return `[0, 0, 0, 0]` transparent black instead of transparent white.